### DISCUSSION TITLE
How to link fit() and fit_transform() together?

### DISCUSSION THREAD

**Discussion by Xtina0** [casact/chainladder-python#450] [General]
Hi there,

I noticed that only fit_transform() has link_ratio as an attribute whereas fit() doesnt.  
- Would you be able to pin point where in the Chainladder package fit_tranform().link_ratio is produced? 
- I also have trouble understanding how fit_transform() links back with fit() (if at all) i.e. what functionalities do they share? Could you please elaborate on this? 

Many thanks,
Xtina

**Comment by jbogaardt** [comment]
You might be comingling the concepts of Estimators and Triangles. Triangles have a `link_ratio` property, estimators do not.  Estimators are all other classes that estimate something from a triangle such as `Development`, `BornhuetterFerguson`, `TailCurve`, etc.

`fit`, `transform` and `fit_transform`  function in the same way they work in `sklearn`.  

The `fit` method returns the model you are working on with fitted attributes. `transform` always returns a `Triangle` object augmented with any additional info passed on by the model you are working with.

So if you're using a `Development` model, and only call fit. The resulting object will be the `Development` estimator itself and will have all the `Development` fitted attributes, such as `ldf_`, cdf_`, etc. If you transform a Triangle, then those attributes will be passed on to the triangle, along with the attributes the triangle already had, such as `link_ratio`.

`fit_transform` is a shorthand for the two steps of fitting and transforming.  That is
```python
 Development().fit(X).transform(X)
# can be expressed as
Development().fit_transform(X)
```

Does that clarify?

**Reply by Xtina0** [reply]
Yes and thanks for your clarification! 
If I may have two follow-up questions - 
1. I am wondering where in the code controls the rounding (i.e. s.f.). For example, in **Development** model?  Specifically I would like to make use of ldf_ but with a slightly different rounding. 
2. While I have your attention here, Would you be able to elaborate on how each folder links to each other below? Appreciate this might be a rather vague question, I just want to have a better understanding of the code set up so I can make changes.  
![image](https://github.com/casact/chainladder-python/assets/139907690/350106d3-6ce7-4d70-a7ee-28da155a0f82)

**Reply by jbogaardt** [reply]
1. There is no internal rounding on any calculations in chainladder. The display may show formatted rounding and that is [here](https://github.com/casact/chainladder-python/blob/master/chainladder/core/display.py#L56-L64). If you want to change the precision of actual data underlying Triangle, just call its `round` method, e.g. `cl.load_sample('raa').link_ratio.round(1)` which works the same way as [pandas.DataFrame.round](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.round.html). 
2. The folders correspond 1-to-1 with the [User Guide](https://chainladder-python.readthedocs.io/en/latest/user_guide/index.html). The only renaming going on is that the `core` folder contains the `Triangle` object.
